src/Swift.Runtime/src/Library/ARC.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// ARC is a class containing p/invokes for Swift Automatic Reference Counting and memory management.
+/// </summary>
+public static class Arc {
+    const string swiftCore = "/usr/lib/swift/libswiftCore.dylib";
+
+    /// <summary>
+    /// Retain a heap-allocated Swift object
+    /// </summary>
+    /// <param name="p">Pointer to an unmanaged Swift object, must be non-null.</param>
+    [DllImport(swiftCore)]
```

Comment:
Especially in the swift related code I think it would be good to be explicit about calling conventions. To me it's surprising this uses the C calling convention.